src/modules/0.0.18/aws_cloudfront/index.ts

```diff
@@ -0,0 +1,323 @@
+import {
+    CloudFront,
+    DistributionConfig,
+    paginateListDistributions,
+    waitUntilDistributionDeployed,
+    DistributionSummary,
+    GetDistributionCommandOutput,
+} from '@aws-sdk/client-cloudfront'
+
+import { AWS, crudBuilder2, paginateBuilder, } from '../../../services/aws_macros'
+import { Distribution, viewerProtocolPolicyEnum, } from './entity'
+import { Context, Crud2, MapperBase, ModuleBase, } from '../../interfaces'
+import { WaiterOptions } from '@aws-sdk/util-waiter'
+import isEqual from 'lodash.isequal'
+import _, { forEach } from 'lodash'
+
+class DistributionMapper extends MapperBase<Distribution> {
+    module: AwsCloudfrontModule;
+    entity = Distribution;
+    equals = (a: Distribution, b: Distribution) => {
+      // specific origins comparison
+      const originsA = a.origins.map(({Id, DomainName}) => ({Id, DomainName}));
+      const originsB = b.origins.map(({Id, DomainName}) => ({Id, DomainName}));
```

Comment:
Won't the later `isEqual` call depend on origin ordering in these output arrays?

Perhaps it'd be better to do something like:

```suggestion
      const originsA = Object.fromEntries(a.origins.map(({Id, DomainName}) => ([Id, DomainName])));
      const originsB = Object.fromEntries(b.origins.map(({Id, DomainName}) => ([Id, DomainName])));
```

This turns it into objects where the keys are the `Id` values` and the values are the `DomainName` values, and object comparison doesn't care about key ordering, I am pretty sure.

Reply:
i wasn't aware that lodash was checking the order to compare, good catch